spec/controllers/queue_items_controller_spec.rb

```diff
@@ -83,6 +83,161 @@
     end
   end
 
+  describe 'PATCH update' do
+    context 'with authenticated user' do
+
+      let(:current_user) { Fabricate(:user) }
+      let(:video1) { Fabricate(:video) }
+      let(:video2) { Fabricate(:video) }
+      let(:item1) { Fabricate(:queue_item, queue_position: -1) }
+      let(:item2) { Fabricate(:queue_item, queue_position: -1) }
+
+      before do
+        session[:user_id] = current_user.id
+
+        item1.update_attributes(user: current_user, video: video1)
+        item2.update_attributes(user: current_user, video: video2)
+
+        item1.rating = 1
+        item2.rating = 2
+      end
+
+      context 'updating queue positions' do
+        it 'redirects to my_queue_path for invalid positions' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 1.5, rating: 1}]
+
+          expect(response).to redirect_to my_queue_path
+        end
+
+        it 'sets the flash-error for invalid positions' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 1.5, rating: 1}
+          ]
+
+          expect(flash[:error]).not_to be_nil
+        end
+
+        it 'updates positions so that the lowest position is 1' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 2, rating: 1},
+            {id: item2.id, position: 4, rating: 1}
+          ]
+          expect(QueueItem.find(item1.id).queue_position).to eq 1
+        end
+
+        it 'reorders positions so that the lowest position is 1' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 3, rating: 1},
+            {id: item2.id, position: 2, rating: 1}
+          ]
+          expect(QueueItem.find(item2.id).queue_position).to eq 1
+        end
+
+        it 'reorders so that all positions are contiguous' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 3, rating: 1},
+            {id: item2.id, position: 2, rating: 1}
+          ]
+          expect(QueueItem.find(item1.id).queue_position).to eq 2
+          expect(QueueItem.find(item2.id).queue_position).to eq 1
+        end
+
+        it 'updates all positions in a transaction' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 3, rating: 1},
+            {id: item2.id, position: 2.5, rating: 1}
+          ]
+          expect(QueueItem.find(item1.id).queue_position).to eq -1
+          expect(QueueItem.find(item2.id).queue_position).to eq -1
+        end
+
+        it 'redirects to my_queue after successful update' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 1, rating: 1},
+            {id: item2.id, position: 2, rating: 1}
+          ]
+          expect(response).to redirect_to my_queue_path
+        end
+
+        it 'does not update queue when position-data is incomplete' do
```

Comment:
This spec and the spec on line 146 are basically testing the same thing. You don't need both. I prefer this one over the on line 146.